### DISCUSSION TITLE
Do you have any solution if I need to update the data everyday?

### DISCUSSION THREAD

**Discussion by CrazyTianC** [milvus-io/milvus#4843] [Q&A and General discussion]
I am doing the job about the recommendation.
In my situation, everyday, I'll recalculate the vector for each user, and insert them into a milvus collection. I'll delete the old one, and build the index of the new collection. So that, when I search, I can use the latest data.
The problem is that, when the old one is deleted while the new one is not yet builded, I can't search anything.
Do you have any solution about this situation? Is it better to use collection or partition? 

Maybe this discussion is not about the milvus, look forward to your reply.

**Comment by yhmo** [comment]
My understood of your situation:
- Assume you have 1M users.
- On day 1, you calculate 1M vectors from the users, create a collection "day_1", insert 1M vectors into this collection, build index for "day_1".
- On day 2, you recalculate 1M vectors, delete the "day_1", create a new collection "day_2", insert 1M vectors into "day_2", build index for "day_2"
- While the index is not finished, you perform search to "day_2", you can't get any result from search, right?

In milvus, if a collection has no index, it will do brute-force search. So, no matter whether the collection has index, the search should return a result.

**Reply by CrazyTianC** [reply]
your understood is right. But if it does the brute-force search, how to guarantee the performance? Does it have any mechanism to return the result in a limited time, even it does not search for every vector?

**Reply by yhmo** [reply]
How many vectors in the collection? If vector count is not large, the performance (index and non-index) is almost equal. 
If the vector count is large, you can build index in night.
